### ISSUE TITLE
Debian paperwork package

### ISSUE THREAD

**Issue by fmitha** [jflesch/paperwork#466]
I'm opening an issue for Debian packaging of paperwork. I'm aware of an RFP, namely
https://bugs.debian.org/721287, but not of any activity on this front. This software looks like it would
not be hard to package (famous last words, perhaps). Are the developers interested in and supportive of a Debian package? What about the users?

Regards, Faheem Mitha

**Comment by jflesch** [comment]
I made calls for Debian maintainers many times [on a website where I go often](http://linuxfr.org/). Up to now, about 3 people have tried to do the job. All of them ran out of free time before finishing. However, it was when 0.2.x was the current version. With 0.3.x the dependencies should be much more lightweight. So I guess it should be worth retrying.

I recently discovered that @plietar got [PyOCR](https://github.com/jflesch/pyocr) packaged in Debian sid (which makes me really really happy since I have been a Debian user for about 10 years now :D). I can only hope he is now working on getting [Pyinsane](https://github.com/jflesch/pyinsane) and Paperwork 0.3.x packaged as well.

**Comment by fmitha** [comment]
Hi jflesch,

Is there any existing packaging available online, at Github or anywhere else?
The RFP has nothing, and I don't see any ITPs filed either. Anyone who is aware of this, please put links as appropriate.

I've never used Paperwork. I just read the LWN article about it just now. But I do
use `gscan2pdf`, which is similar to Paperwork, I think. And alternatives are worth trying.

I'll initially try the quick and dirty approach described in https://github.com/jflesch/paperwork/blob/stable/doc/install.debian.markdown
and see what I think.

**Comment by jflesch** [comment]
Afaik, there is no .deb packaging that is up-to-date. (someone made basic/maybe-incomplete Paperwork packages a while ago, but I can't find them back)

Regarding gscan2pdf, I guess you already know it, but the goal is not really the same than Paperwork. Gscan2pdf simply generates PDF from scans and let you use other piece of software to index and search your PDFs. It gives you a lot of options to get PDF the exact way you want, with the best OCR result possible.

On the other hand, Paperwork acts as a full personal document manager (index + search). Because it manages indexing + searching, it can ensure fuzzy searches are possible. Which means OCR doesn't have to be perfect. It just has to be good-enough.
So Paperwork doesn't give you a lot of options to fine-tune scan+OCR, and it's actually by design.

I guess both approaches make sense, depending on what you want to do.

**Comment by fmitha** [comment]
Hi jflesch,

I see. Thank you for the explanation. 

It would be helpful to get a status update from Paul Lietar.
Also, what did Ross Vandegrift end up doing, if anything?

**Comment by jflesch** [comment]
Last news I had from @rvandegrift was [06/14/2014](https://groups.google.com/d/msg/paperwork-gui/gHUyna_6ZlE/9YlSn0JyaekJ), when he told us that it couldn't work on it anymore.

He had a Debian repository running with Paperwork 0.1.2. Since they haven't been checked by anyone else from Debian, I'm not sure if they are complete/correct.

**Comment by fmitha** [comment]
Ok. Some indication of the next steps to take here would be helpful. Is pyinsane a dependency of Paperwork? It does not seem to be in Debian, so would it need to be packaged first?

I'm not a Debian Maintainer or Developer, but I'm an experienced Debian user, have some packaging experience, and am a long term Python user. I'm also [active on Stack Exchange](https://unix.stackexchange.com/users/4671/faheem-mitha?tab=profile).

Anyway, let me give Paperwork a try.

**Comment by plietar** [comment]
Hi,
Sorry I stopped working on this without any notice.
I based my work off Ross'.
Everything is on my Github repositories, on the `debian` branches of each project :
https://github.com/plietar/paperwork
https://github.com/plietar/pyocr
https://github.com/plietar/pyinsane

The last packaged versions are paperwork 0.2.2, pyocr 0.3.0 and pyinsane 1.3.8.
They were pretty much complete. Unfortunately the process to upload into the Debian archive is quite long and tedious, and both me and the DD supervising the upload got busy with other stuff.

**Comment by fmitha** [comment]
Hi  plietar,

Great. I see pyocr is already there, with you listed as Maintainer.

> They were pretty much complete.

Does that mean they are more or less ready to be included in Debian, modulo version numbers?
In any case, I'll give these packages a whirl.

                                                                         Regards, Faheem Mitha

**Comment by fmitha** [comment]
I got a segfault building the pyinsane package. Maybe that's a pysegfault?
[python-pyinsane_1.3.8-1_amd64.build.txt](https://github.com/jflesch/paperwork/files/172393/python-pyinsane_1.3.8-1_amd64.build.txt). 
(Github wouldn't upload it unless I added the `.txt`).
This also seems to be the most recent release. Should I report this somewhere?

**Comment by jflesch** [comment]
Yes, [here](https://github.com/jflesch/pyinsane/issues). But please do get a stack trace with gdb first. Otherwise I won't be able to do anything about it.

```
$ git clone git@github.com:jflesch/pyinsane.git
$ cd pyinsane
$ gdb --args python ./run_tests.py
> r
(...)
(<segfault>)
> bt
```

**Comment by jflesch** [comment]
Hm, actually, I'm not even sure that the Pyinsane tests are the one segfaulting here ...

**Comment by jflesch** [comment]
By the way, if I remember correctly, those tests require the user to have a scanner connected. I don't think it's a good idea to run them in the build script.

**Comment by fmitha** [comment]
> But please do get a stack trace with gdb first.

Ok, will do.

> By the way, if I remember correctly, those tests require the user to have a scanner connected. I don't think it's a good idea to run them in the build script.

I've currently got two scanners connected.

**Comment by jflesch** [comment]
> I've currently got two scanners connected.

My point is, builds can be done on remote build machine. In that case, you don't want those tests to run. So it shouldn't be part of the build file.

**Comment by plietar** [comment]
The pyinsane test creates a dummy device if no physical scanner is present.
https://github.com/jflesch/pyinsane/blob/master/tests/tests_abstract.py#L21

I did all my builds in a VM, without a scanner

**Comment by jflesch** [comment]
Oh right, I forgot this contrib :). Sorry.

**Comment by fmitha** [comment]
I disabled the pyinsane tests and built the package, and then Paperwork.

When running Paperwork, I see:
```
*** BUG ***
In pixman_region32_init_rect: Invalid rectangle passed
Set a breakpoint on '_pixman_log_error' to debug
```
on the console. Is this normal? Should I report it?

**Comment by jflesch** [comment]
I get this warning too. But things have been working fine.
Not sure where it comes from.

**Comment by jflesch** [comment]
Update: the warning came from invalid calls to Gtk.Widget.queue_draw() (I passed negative values sometimes). It has been fixed in the 1.0.

Is this ticket still useful, or can  I close it ?

**Comment by fmitha** [comment]
Hi @jflesch,

This issue is about Debian packaging of `paperwork`. It doesn't look like that packaging has happened yet, so it seems reasonable to leave it open. But it's up to you.

I see something about getting a segfault building the package, and you then requested a gdb stack trace. But it doesn't look like I provided one. Is this still something forth following up on?

**Comment by jflesch** [comment]
There have been a *lot* of changes in Pyinsane since. So it's worth trying again and see if it segfaults again

**Comment by jflesch** [comment]
ping ?

**Comment by fmitha** [comment]
Sorry, jflesch,

This issue is currently way down my priority list, and I'm not sure when I'll get a chance to take a look at it. Feel free to close if you wish.

**Comment by jflesch** [comment]
No problem.
I guess I'll keep it open for now. Maybe at some point I'll make the packages myself ....

**Comment by jflesch** [comment]
There is now a Flatpak package available: https://github.com/openpaperwork/paperwork/tree/unstable/flatpak#readme
However it would still be nice to have a clean Debian packaging.
